src/ac_power_flow_residual.jl

```diff
@@ -40,24 +46,68 @@ Create an instance of `ACPowerFlowResidual` for a given time step.
 and net bus reactive power injections.
 """
 function ACPowerFlowResidual(data::ACPowerFlowData, time_step::Int64)
+    is_ref(bt::Val{PSY.ACBusTypes.REF}) = true
```

Comment:
Unconventional to put functions inside of functions

Reply:
not sure what this does to performance

Reply:
TODO: remove the functions is_ref, is_refpv  after the @scoped_enum is improved

This PR already addresses it: https://github.com/NREL-Sienna/InfrastructureSystems.jl/pull/458